src/core/Rule.ml

```diff
@@ -275,6 +275,52 @@ and extract_transform = NoTransform | Unquote | ConcatJsonArray
 *)
 and extract_reduction = Separate | Concat [@@deriving show]
 
+(*****************************************************************************)
+(* secrets mode *)
+(*****************************************************************************)
+
+(* This type encodes a basic HTTP request; mainly used for in the secrets
```

Comment:
Great comment!
Also might be useful to explain why we don't reuse the request type of http libraries like cohttp, which have their own request type.